yt/data_objects/construction_data_containers.py

```diff
@@ -2303,59 +2314,56 @@ def _sanitize_edge(self, edge, default):
             edge_units = edge.units
         else:
             edge_units = 'code_length'
-        return self.ds.arr(edge, edge_units)
+        return 1.0*self.ds.arr(edge, edge_units, dtype="float64")
```

Comment:
so this 1.0 is a complete fudge. However, if I don't include it then 
`self.ds.arr([0, 0, 0], dtype="float64")` seems to revert to `int64`.
Any ideas what causes this?

Reply:
I *assume* but am not certain that this is because `unyt` tries to allow things to be in any type, and maybe (?) there's a logical check that doesn't catch the `dtype`?